File: tests/fixtures/smoke/lua/sample.lua
-- Smoke-tier fixture (BACK-431 Issue G) — Lua.
local function validate(order)
    if order == nil then
        error("empty order")
    end
    return order
end

local function processOrder(order)
    local ok, result = pcall(validate, order)
    if not ok then
        return nil
    end
    local count = 0
    while count < 3 do
        count = count + 1
    end
    return string.upper(result)
end

local function run(order)
    return processOrder(order)
end

File: tests/fixtures/smoke/zig/sample.zig
// Smoke-tier fixture (BACK-431 Issue G) — Zig.
const std = @import("std");

fn validate(order: ?[]const u8) ![]const u8 {
    if (order == null) {
        return error.EmptyOrder;
    }
    return order.?;
}

fn processOrder(order: ?[]const u8) ?[]const u8 {
    const result = validate(order) catch {
        return null;
    };
    var count: u32 = 0;
    while (count < 3) {
        count += 1;
    }
    return result;
}

fn run(order: ?[]const u8) ?[]const u8 {
    return processOrder(order);
}
